Prototyping/Sit/sit.js

```diff
@@ -0,0 +1,692 @@
+                    str = i + 1;
+                    overlayPreSitLoaded[i] = TextureCache.prefetch(OVERLAY_PRESIT_URL_ROOT + str + OVERLAY_PRESIT_URL_POSTFIX);
+                }
+                OVERLAY_PRESIT_URL_TEXT = TextureCache.prefetch(OVERLAY_PRESIT_URL_TEXT);
+            },
+            create: function () {
+                if (overlayPreSit) {
+                    return;
+                }
+
+                overlayPreSit = Overlays.addOverlay("image3d", {
+                    position: Vec3.sum(Camera.position, Vec3.multiplyQbyV(Camera.orientation, { x: 0, y: 0.1, z: -1 })),
+                    rotation: Camera.orientation,
+                    parentId: MyAvatar.sessionUUID,
+                    parentJoint: "Head",
+                    dimensions: { x: 0.2, y: 0.2 },
+                    url: overlayPreSitLoaded[index].url,
+                    ignoreRayIntersection: false,
+                    drawInFront: true,
+                    visible: true,
+                    emissive: true
+                });
+
+                overlayPreSitText = Overlays.addOverlay("image3d", {
+                    position: Vec3.sum(Camera.position, Vec3.multiplyQbyV(Camera.orientation, { x: 0, y: -0.05, z: -1 })),
+                    rotation: Camera.orientation,
+                    parentId: MyAvatar.sessionUUID,
+                    parentJoint: "Head",
+                    dimensions: { x: 0.425, y: 0.425 },
+                    url: OVERLAY_PRESIT_URL_TEXT.url,
+                    ignoreRayIntersection: false,
+                    drawInFront: true,
+                    visible: true,
+                    emissive: true
+                });
+
+                overlayPreSitStart = Date.now();
+            },
+            remove: function () {
+                if (overlayPreSit !== null) {
+                    Overlays.deleteOverlay(overlayPreSit);
+                    overlayPreSit = null;
+                    overlayPreSitStart = 0;
+                    index = 0;
+
+                    if (overlayPreSitText !== null) {
+                        Overlays.deleteOverlay(overlayPreSitText);
+                    }
+                }
+            },
+            update: function () {
+
+                var now = Date.now();
+
+                if (index >= overlayPreSitLoaded.length) {
+                    return;
+                }
+
+                var cameraRotation = Camera.orientation;
+
+                if (overlayPreSit) {
+
+                    var updateObj = {
+                        rotation: cameraRotation,
+                        position: Vec3.sum(Camera.position, Vec3.multiplyQbyV(Camera.orientation, { x: 0, y: 0.1, z: -1 })),
+                    }
+
+                    var timePassed = now - overlayPreSitStart;
+                    if (timePassed >= OVERLAY_PRESIT_FRAME_DURATION) {
+                        overlayPreSitStart = now;
+                        index = index + 1;
```

Comment:
Please give this variable a more descriptive name, I'm concerned that someone might inadvertently change this, because we use index as a variable in several other places in this file.  How about preSitLoadIndex?